### PR TITLE
Add point cloud destination parameter and output to Processing

### PR DESCRIPTION
**Pull request by alexbruy**
## Description

Extend and improve point clouds support in Processing, namely:

- fix filtering of point cloud layers in `QgsProcessingMapLayerComboBox`-based widget used for parameters
- add destination point cloud parameter and point cloud output with corresponding widget wrapper. At the moment they are useful for 3rd party providers which create point cloud outputs, as allow them to load results into canvas. Also, as there are plans to add native point cloud algorithms at some point having all the infrastructure ready is important.
- expose newly added output and destination parameter to `algfactory` wrapper

### CODE REVIEW

File: src/gui/processing/qgsprocessingguiregistry.cpp [hunk 1]
```diff
@@ -81,6 +81,8 @@ QgsProcessingGuiRegistry::QgsProcessingGuiRegistry()
   addParameterWidgetFactory( new QgsProcessingMeshDatasetTimeWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingPointCloudLayerWidgetWrapper() );
   addParameterWidgetFactory( new QgsProcessingAnnotationLayerWidgetWrapper() );
+  addParameterWidgetFactory( new QgsProcessingAnnotationLayerWidgetWrapper() );
```
Comment (nyalldawson) [inline src/gui/processing/qgsprocessingguiregistry.cpp:84]:
```suggestion
```

**Review by nyalldawson — COMMENTED** [review]
Looking good !

File: python/processing/algfactory.py [hunk 2]
```diff
@@ -320,10 +322,12 @@ class ProcessingAlgFactory():
     RASTER_LAYER = "RASTER_LAYER",
     VECTOR_LAYER = "VECTOR_LAYER",
     MESH_LAYER = "MESH_LAYER",
+    POINT_CLOUD_LAYER = "MESH_LAYER",
```
Comment (nyalldawson) [inline python/processing/algfactory.py:325]:
```suggestion
    POINT_CLOUD_LAYER = "POINT_CLOUD_LAYER",
```

File: python/processing/algfactory.py [hunk 3]
```diff
@@ -320,10 +322,12 @@ class ProcessingAlgFactory():
...
     FILE_DEST = "FILE_DEST",
     FOLDER_DEST = "FOLDER_DEST",
     RASTER_LAYER_DEST = "RASTER_LAYER_DEST",
     VECTOR_LAYER_DEST = "VECTOR_LAYER_DEST",
+    POINTCLOUD_LAYER_DEST = "VECTOR_LAYER_DEST",
```
Comment (nyalldawson) [inline python/processing/algfactory.py:330]:
```suggestion
    POINTCLOUD_LAYER_DEST = "POINTCLOUD_LAYER_DEST",
```

File: src/core/processing/qgsprocessingprovider.cpp [hunk 4]
```diff
@@ -185,6 +190,18 @@ bool QgsProcessingProvider::isSupportedOutputValue( const QVariant &outputValue,
     }
     return true;
   }
+  else if ( parameter->type() == QgsProcessingParameterPointCloudDestination::typeName() )
+  {
+    const QFileInfo fi( outputPath );
+    const QString extension = fi.completeSuffix();
+    qDebug() << supportedOutputPointCloudLayerExtensions();
```
Comment (nyalldawson) [inline src/core/processing/qgsprocessingprovider.cpp:197]:
```suggestion
```

File: python/processing/algfactory.py (hunk 2 above)
Comment (alexbruy) [reply python/processing/algfactory.py]: Ouch, sorry. I really should not code with that nasty cold.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**